back/engines/commercial/idea_custom_fields/app/controllers/idea_custom_fields/web_api/v1/admin/idea_custom_fields_controller.rb

```diff
@@ -173,36 +176,41 @@ def update_field(&_block)
     end
 
     def update_options(field, options_params, errors, field_index)
-      options = field.options
-      options_by_id = options.index_by(&:id)
-      given_ids = options_params.pluck :id
+      {}.tap do |option_temp_ids_to_ids_mapping|
+        options = field.options
+        options_by_id = options.index_by(&:id)
+        given_ids = options_params.pluck :id
 
-      deleted_options = options.reject { |option| given_ids.include? option.id }
-      deleted_options.each do |option|
-        SideFxCustomFieldOptionService.new.before_destroy option, current_user
-        option.destroy!
-        SideFxCustomFieldOptionService.new.after_destroy option, current_user
-      end
-      options_params.each_with_index do |option_params, option_index|
-        if option_params[:id]
-          option = options_by_id[option_params[:id]]
-          option.assign_attributes option_params
-          SideFxCustomFieldOptionService.new.before_update option, current_user
-          unless option.save
-            add_options_errors option.errors.details, errors, field_index, option_index
-            next
-          end
-          SideFxCustomFieldOptionService.new.after_update option, current_user
-        else
-          option = CustomFieldOption.new option_params.merge(custom_field: field)
-          SideFxCustomFieldOptionService.new.before_create option, current_user
-          unless option.save
-            add_options_errors option.errors.details, errors, field_index, option_index
-            next
+        deleted_options = options.reject { |option| given_ids.include? option.id }
+        deleted_options.each do |option|
+          SideFxCustomFieldOptionService.new.before_destroy option, current_user
+          option.destroy!
+          SideFxCustomFieldOptionService.new.after_destroy option, current_user
+        end
+        options_params.each_with_index do |option_params, option_index|
+          if option_params[:id]
+            option = options_by_id[option_params[:id]]
+            option.assign_attributes option_params
+            SideFxCustomFieldOptionService.new.before_update option, current_user
+            unless option.save
+              add_options_errors option.errors.details, errors, field_index, option_index
+              next
+            end
+            SideFxCustomFieldOptionService.new.after_update option, current_user
+          else
+            update_params = option_params.except('temp_id')
```

Comment:
I think we're creating a new option here, so I would suggest renaming `update_params` to `create_params`.

Reply:
Changed.